WallsAndHoles/tiletemplate.cpp

```diff
@@ -57,3 +77,22 @@ void TileTemplate::setColor(QColor color)
     emit exclusivePropertyChanged();
     emit changed();
 }
+
+void TileTemplate::setMaterial(std::unique_ptr<SimpleTexturedMaterial> &&material)
+{
+    mMaterial = std::move(material);
+
+    emit materialChanged();
```

Comment:
I'd have to investigate this a little further to know for sure, but I'm pretty sure emit changed() will cause the mesh to be remade, but that doesn't need to happen. Just noting here to make sure only the required signals are emitted.